src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  NavControllerMock,
+  NavParamsMock,
+  ConfigMock,
+  PlatformMock,
+  ModalControllerMock,
+} from 'ionic-mocks';
+import { MockComponent } from 'ng-mocks';
+
+import { AppModule } from '../../../../app/app.module';
+import { TestReportCatDPage } from '../test-report.cat-d.page';
+import { AuthenticationProvider } from '../../../../providers/authentication/authentication';
+import { AuthenticationProviderMock } from '../../../../providers/authentication/__mocks__/authentication.mock';
+import { CompetencyComponent } from '../../components/competency/competency';
+import { CompetencyButtonComponent } from '../../components/competency-button/competency-button';
+import { DateTimeProvider } from '../../../../providers/date-time/date-time';
+import { DateTimeProviderMock } from '../../../../providers/date-time/__mocks__/date-time.mock';
+import { DrivingFaultSummaryComponent } from '../../components/driving-fault-summary/driving-fault-summary';
+import { TickIndicatorComponent } from '../../../../components/common/tick-indicator/tick-indicator';
+import { ToolbarComponent } from '../../components/toolbar/toolbar';
+import { By } from '@angular/platform-browser';
+import { StoreModule } from '@ngrx/store';
+import { testReportReducer } from '../../test-report.reducer';
+import { LegalRequirementComponent } from '../../components/legal-requirement/legal-requirement';
+import { EtaComponent } from '../../components/examiner-takes-action/eta';
+// TODO Implement Cat D TestData Reducer
+import { initialState } from '../../../../modules/tests/test-data/cat-c/test-data.cat-c.reducer';
+import { EcoComponent } from '../../components/eco/eco';
+import { TestReportValidatorProvider } from '../../../../providers/test-report-validator/test-report-validator';
+import {
+  TestReportValidatorProviderMock,
+} from '../../../../providers/test-report-validator/__mocks__/test-report-validator.mock';
+import { ModalEvent } from '../../test-report.constants';
+import { CAT_D } from '../../../page-names.constants';
+import { NavigationStateProvider } from '../../../../providers/navigation-state/navigation-state';
+import { NavigationStateProviderMock } from '../../../../providers/navigation-state/__mocks__/navigation-state.mock';
+import { candidateMock } from '../../../../modules/tests/__mocks__/tests.mock';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import { configureTestSuite } from 'ng-bullet';
+// TODO Implement Cat D UncoupleRecoupleComponent
+import { UncoupleRecoupleComponent } from '../../cat-c/components/uncouple-recouple/uncouple-recouple';
+// TODO Implement Cat D ReverseLeftComponent
+import { ReverseLeftComponent } from '../../cat-c/components/reverse-left/reverse-left';
+// TODO Implement Cat D ReverseLeftPopoverComponent
+import { ReverseLeftPopoverComponent } from '../../cat-c/components/reverse-left-popover/reverse-left-popover';
```

Comment:
These should be copied over as part of the pr the category specific components are part of the page.

Reply:
Uncouple/Recouple and VehicleChecksComponent haven't been created yet, although the Cat D for Reversing gets copied over in the next PR

Reply:
they should be included in this PR, The copy is of the page and it's custom component.